app/models/git/repository.rb

```diff
@@ -20,6 +20,8 @@ def initialize(repo_url: nil)
         @repo_url = repo_url || "https://github.com/exercism/#{repo_name}"
       end
 
+      @branch_ref = branch_ref || ENV.fetch("GIT_CONTENT_BRANCH", MAIN_BRANCH_REF)
```

Comment:
```suggestion
      @branch_ref = branch_ref || ENV["GIT_CONTENT_BRANCH"].presence || MAIN_BRANCH_REF
```

We need to use `presence` here to guard against an empty string. Fetch only checks for `nil`.